src/kernels/kernel.ts

```diff
@@ -53,13 +58,41 @@ import { Cancellation, isCancellationError } from '../platform/common/cancellati
 import { KernelProgressReporter } from '../platform/progress/kernelProgressReporter';
 import { DisplayOptions } from './displayOptions';
 import { SilentExecutionErrorOptions } from './helpers';
+import dedent from 'dedent';
+import { IAnyMessageArgs } from '@jupyterlab/services/lib/kernel/kernel';
+
+const widgetVersionOutPrefix = 'e976ee50-99ed-4aba-9b6b-9dcd5634d07d:IPyWidgets:';
```

Comment:
what is this value? where is it from?

Reply:
its just a GUUID, you can see how it is used, in print statements.
see here (further below)
```typescript
        try:
            import ipywidgets as _VSCODE_ipywidgets
            print("${widgetVersionOutPrefix}" + _VSCODE_ipywidgets.__version__)
            del _VSCODE_ipywidgets
        except:
            pass
        `;

            const version = await this.executeSilently(session, [codeToDetermineIPyWidgetsVersion]).catch((ex) =>
                traceError('Failed to determine version of IPyWidgets', ex)
            );
            traceError('Determined IPyKernel Version', JSON.stringify(version));
            if (Array.isArray(version)) {
                const isVersion8 = version.some((output) =>
                    (output.text || '')?.toString().includes(`${widgetVersionOutPrefix}8.`)
                );
                const isVersion7 = version.some((output) =>
                    (output.text || '')?.toString().includes(`${widgetVersionOutPrefix}7.`)
                );
```